### PR TITLE
Update Find for Illness

### PR DESCRIPTION
**Pull request by raysonchia**
An update to Find based on the new Illness constraints in #66

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -208,11 +208,12 @@ Format: `find [ic/NRIC] [n/NAME] [p/PHONE_NUMBER] [e/EMAIL] [g/GENDER] [b/BIRTHD
...
 * Only full words will be matched e.g. `Han` will not match to `Hans`
 * Patients matching at least one keyword for a specific parameter will be returned (i.e. `OR` search). e.g. `n/Hans Bo` will return patients named `Hans Gruber`, `Bo Yang`
-* Multiple search parameters can be combined to narrow down search results (i.e. `AND` search). e.g. `n/taylor g/m` will return male patients named Taylor
+* Multiple search parameters can be combined to narrow down search results (i.e. `AND` search). e.g. `n/taylor g/m` will return all `Male` patients named `Taylor`
 
 Examples:
 * `find n/John` returns `john` and `John Doe`
-* `find d/paracetamol i/infectious disease` returns all patients with paracetamol allergy and infectious disease
+* `find d/paracetamol i/infectious` returns all patients with a `Paracetamol` allergy who also have the `Infectious Diseases` illness category
+* `find i/diseases others` returns all patients with `Infectious Diseases`, `Degenerative Diseases` and `Others` illness categories
```
Comment (sebin0817) [inline docs/UserGuide.md:216]: Please remember to update the `.puml` file to show the modified patient class

File: src/test/java/seedu/address/logic/parser/ParserUtilTest.java [hunk 2]
```diff
@@ -50,7 +50,7 @@ public class ParserUtilTest {
     private static final String VALID_PHONE = "65435623";
     private static final String VALID_EMAIL = "rachel@example.com";
     private static final String VALID_DRUG_ALLERGY = "Penicillin";
-    private static final String VALID_ILLNESS_1 = "Infectious Disease";
+    private static final String VALID_ILLNESS_1 = "Infectious Diseases";
```
Comment (jmsandiegoo) [inline src/test/java/seedu/address/logic/parser/ParserUtilTest.java:53]: Good job for spotting a typo bug here 👍

File: src/main/java/seedu/address/model/person/illness/Illness.java [hunk 3]
```diff
@@ -17,12 +18,28 @@ public class Illness {
             + "(You could type just the first word / letters for convenience "
             + "as long it corresponds to a single category.): \n"
             + "Infectious Disease, Chronic Conditions, Autoimmune Disorders, Genetic Disorders, \n"
-            + "Mental Disorders, Neurological Disorders, Metabolic Disorder, Nutritional Deficiencies, \n"
+            + "Mental Disorders, Neurological Disorders, Metabolic Disorders, Nutritional Deficiencies, \n"
```
Comment (jmsandiegoo) [inline src/main/java/seedu/address/model/person/illness/Illness.java:21]: 👍 for another typo fix!

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 4]
```diff
@@ -188,6 +188,21 @@ public static Set<Illness> parseIllnesses(Collection<String> illnesses) throws P
...
+     */
+    public static String parseFindIllness(String illness) throws ParseException {
+        requireNonNull(illness);
+        String trimmedIllness = illness.trim();
+        if (!Illness.isValidIllnessFind(trimmedIllness)) {
+            throw new ParseException(Illness.FIND_MESSAGE_CONSTRAINTS);
+        }
+        return trimmedIllness;
+    }
+
```
Comment (jmsandiegoo) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:205]: Awesome! Seems like you have improved the parsing logic to not create a new Illness logic with an invalid IllnessName. 💯

**Review by jmsandiegoo — APPROVED** [review]
LGTM

**Review by sebin0817 — APPROVED** [review]
LGTM

File: src/test/java/seedu/address/model/person/illness/IllnessTest.java [hunk 5]
```diff
@@ -1,10 +1,15 @@
...
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
 import static seedu.address.testutil.Assert.assertThrows;
 
 import org.junit.jupiter.api.Test;
 
 public class IllnessTest {
+    private static final String VALID_ILLNESS_1 = "Infectious Diseases";
+    private static final String VALID_ILLNESS_2 = "Chronic Conditions";
+    private static final String INVALID_ILLNESS = "Motion Sickness";
```
[the remark this answers is not in the document]
Comment (raysonchia) [reply src/test/java/seedu/address/model/person/illness/IllnessTest.java:12]: Thanks, I will update this part.

**Pull request merged**